File: week3-AppDev-crud/crud-python/Ex08_Query4b_Read_Journey_Details.py
#!/usr/bin/env python3
from db_connection import Connection
import uuid

print('========================================')
print('Start exercise')

spacecraft_name = 'Crew Dragon Endeavour,SpaceX'
journey_id      = uuid.UUID('230995ee-c697-11ea-b7a1-8c85907c08dd')

try:
    connection = Connection()
    output = connection.session.execute(
        "SELECT * FROM spacecraft_journey_catalog WHERE spacecraft_name=%s AND journey_id=%s",
        [spacecraft_name, journey_id]
    )
    for row in output:
       print('Journey has been found')
       print('- Uid:\t\t ', row.journey_id);
       print('- Spacecraft:\t', row.spacecraft_name);
       print('- Summary:\t', row.summary);
       print('- Active:\t', row.active);
       print('- Takeoff:\t', row.start);
       print('- Landing:\t', row.end);
except Exception as e: 
    print(e)
    print('Failure')
else:

    print('Success')
    print('Closing connection (up to 10s)')
finally:
    connection.close()
print('========================================')

File: week3-AppDev-crud/crud-nodejs/Ex08_Query4b_Read_Journey_Details.js
const connection = require('./db_connection')

console.log("========================================")
console.log("Start exercise")

// Defining our journey
const journey_id      = UUID.fromString('<enter_your_journey_id>')
const spacecraft_name = 'Crew Dragon Endeavour,SpaceX'

// Query
const queryFindJourneyDetails = 'SELECT * FROM spacecraft_journey_catalog WHERE spacecraft_name=? AND journey_id=?'

connection.client.execute(queryFindJourneyDetails, [spacecraft_name, journey_id])
.then(function(result){
    result.rows.forEach(row => {
        console.log('Journey has been found')
        console.log('- Uid:\t\t %s', row.journey_id);
        console.log('- Spacecraft:\t %s', row.spacecraft_name);
        console.log('- Summary:\t %s', row.summary);
        console.log('- Active:\t %s', row.active);
        console.log('- Takeoff:\t %s', row.start);
        console.log('- Landing:\t %s', row.end);
    })
    connection.client.shutdown()
    console.log("SUCCESS")
})
.catch(function(error){
    console.log(error.message)
    connection.client.shutdown()
    console.log("FAILED")
});

console.log("========================================")